hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -332,7 +332,7 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	// Reconcile the HostedControlPlane
-	hcp := controlplaneoperator.HostedControlPlane(controlPlaneNamespace.Name, hcluster.Name)
+	hcp := controlplaneoperator.HostedControlPlane(controlPlaneNamespace.Name, hcluster.Spec.InfraID)
```

Comment:
Why change the name of the HostedControlPlane resource?

Reply:
Because `ensureControlPlane` needs to know the infraID to reconcile the secret expected by CAPI and the name of the HostedControlPlane is something we choose conveniently as we did for the Cluster CR.
Do you have any alternative preference to pass it through?

Reply:
the InfraID is available in the HCP spec https://github.com/openshift/hypershift/blob/9db19cdedea23a14b0d29391c396a20eb752e600/api/v1alpha1/hostedcluster_types.go#L40